DS-102: Expand new-component docs (#1899)
boltdesignsystem/bolt: docs-site/src/pages/docs/50-guides/15-building-a-new-component.md

Bolt supplies a easy to use command that will allow you to quickly create an integrated bolt component. Simply run `yarn cc` in the Bolt repo root and provide answers to the two questions that follow.

The command will create two folders with nested files. One folder is in `docs-site/src/pages/pattern-lab/_patterns/02-components/` and is used exclusively for use with the [pattern library](https://boltdesignsystem.com/pattern-lab/?p=components-overview). The other folder is in `packages/components/` and is the main working directory for your new component.

Once the questions are answered, we recommend running `yarn setup` before beginning development again with `yarn start`.

## Leveraging Other Components

Many times, when creating a new component, it can be useful to leverage another component, for example, the eyebrow text component found nested within [bolt-headline](https://boltdesignsystem.com/pattern-lab/?p=viewall-components-headline). To do so, include the relevant twig markup:

```twig
{% if eyebrow %}
    {% include "@bolt-components-headline/eyebrow.twig" with eyebrow only %}
{% endif %}
```

> By looking at eyebrow's relevant [schema](https://boltdesignsystem.com/pattern-lab/?p=viewall-components-headline), you can see that it accepts an array of data including items variables like `text` and `weight`. This is important when passing data through your component's new twig include on to your leveraged component.

Reviewer: ```suggestion
By looking at the Headline [schema](https://boltdesignsystem.com/pattern-lab/?p=viewall-components-headline), you can see that Eyebrow accepts an array of data including variables like `text` and `weight`. It is important to pass only data specified in the schema to your leveraged component.
```

> Be sure to update the schema in your new component directory to include any relevant fields, and you can even reference your leveraged components like so:

Reviewer: ```suggestion
Be sure to update the schema in your new component directory to include any data you've added, and you can even reference your leveraged components like so:
```
